server/ot_rpc.go

```diff
@@ -132,72 +134,112 @@ func runTranscoder(n *core.LivepeerNode, orchAddr string, capacity int) error {
 func runTranscode(n *core.LivepeerNode, orchAddr string, httpc *http.Client, notify *net.NotifySegment) {
 
 	glog.Infof("Transcoding taskId=%d url=%s", notify.TaskId, notify.Url)
-	var contentType string
+	var contentType, fname string
 	var body bytes.Buffer
+	var tData *core.TranscodeData
 
 	md, err := coreSegMetadata(notify.SegData)
 	if err != nil {
-		glog.Error("Unable to parse segData err=", err)
-		md = &core.SegTranscodingMetadata{} // avoid crash.
+		glog.Errorf("Unable to parse segData taskId=%d url=%s err=%v", notify.TaskId, notify.Url, err)
+		sendTranscodeResult(n, orchAddr, httpc, notify, contentType, &body, tData, err)
+		return
 		// TODO short-circuit error handling
 		// See https://github.com/livepeer/go-livepeer/issues/1518
 	}
 	profiles := md.Profiles
-	md.Fname = notify.Url
+
+	data, err := drivers.GetSegmentData(notify.Url)
+	if err != nil {
+		glog.Errorf("Transcoder cannot get segment from taskId=%d url=%s err=%v", notify.TaskId, notify.Url, err)
+		sendTranscodeResult(n, orchAddr, httpc, notify, contentType, &body, tData, err)
+		return
+	}
+	// Write it to disk
+	if _, err := os.Stat(n.WorkDir); os.IsNotExist(err) {
+		err = os.Mkdir(n.WorkDir, 0700)
+		if err != nil {
+			glog.Errorf("Transcoder cannot create workdir err=%v", err)
+			sendTranscodeResult(n, orchAddr, httpc, notify, contentType, &body, tData, err)
+			return
+		}
+	}
+	// Create input file from segment. Removed after transcoding done
+	fname = path.Join(n.WorkDir, common.RandName()+".tempfile")
+	if err = ioutil.WriteFile(fname, data, 0644); err != nil {
```

Comment:
Wouldn't `0600` be enough, do we need other users to access this file?

Reply:
Fixed